src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/PersonalisationProvider.java

```diff
@@ -225,4 +224,40 @@ private static String readStringCaseField(final AsylumCase asylumCase, final Asy
         final Optional<String> optionalFieldValue = asylumCase.read(caseField, String.class);
         return optionalFieldValue.isPresent() && !optionalFieldValue.get().isEmpty() ? optionalFieldValue.get() : defaultIfNotPresent;
     }
+
+    public String getFtpaDecisionTemplateId(AsylumCase asylumCase,
+                                            GovNotifyTemplateIdConfiguration govNotifyTemplateIdConfiguration, YesOrNo applicant) {
+        FtpaAppellantDecisionOutcomeType ftpaAppellantDecisionOutcomeType = asylumCase
+            .read(FTPA_APPELLANT_DECISION_OUTCOME_TYPE, FtpaAppellantDecisionOutcomeType.class)
+            .orElseThrow(() -> new IllegalStateException("ftpaApplicationDecision is not present"));
+
+        if (ftpaAppellantDecisionOutcomeType.toString().equals(FtpaAppellantDecisionOutcomeType.FTPA_GRANTED.toString())) {
+            return applicant.equals(YesOrNo.YES) ? govNotifyTemplateIdConfiguration.getApplicationGrantedApplicant() : govNotifyTemplateIdConfiguration.getApplicationGrantedOtherParty();
+        } else if (applicant.toString().equals(FtpaAppellantDecisionOutcomeType.FTPA_PARTIALLY_GRANTED.toString())) {
+            return applicant.equals(YesOrNo.YES) ? govNotifyTemplateIdConfiguration.getApplicationPartiallyGrantedApplicant() : govNotifyTemplateIdConfiguration.getApplicationPartiallyGrantedOtherParty();
+        } else if (ftpaAppellantDecisionOutcomeType.toString().equals(FtpaAppellantDecisionOutcomeType.FTPA_REFUSED.toString())) {
+            return applicant.equals(YesOrNo.YES) ? govNotifyTemplateIdConfiguration.getApplicationRefusedApplicant() : govNotifyTemplateIdConfiguration.getApplicationRefusedOtherParty();
+        } else {
+            return applicant.equals(YesOrNo.YES) ? govNotifyTemplateIdConfiguration.getApplicationNotAdmittedApplicant() : govNotifyTemplateIdConfiguration.getApplicationNotAdmittedOtherParty();
+        }
+    }
+
+    public Map<String, String> getFtpaDecisionPersonalisation(AsylumCase asylumCase) {
+        requireNonNull(asylumCase, "asylumCase must not be null");
+
+        return ImmutableMap
+            .<String, String>builder()
+            .put("appealReferenceNumber", asylumCase.read(AsylumCaseDefinition.APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("legalRepReferenceNumber", asylumCase.read(AsylumCaseDefinition.LEGAL_REP_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("homeOfficeReferenceNumber", asylumCase.read(AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("appellantGivenNames", asylumCase.read(AsylumCaseDefinition.APPELLANT_GIVEN_NAMES, String.class).orElse(""))
+            .put("appellantFamilyName", asylumCase.read(AsylumCaseDefinition.APPELLANT_FAMILY_NAME, String.class).orElse(""))
+            .build();
+    }
+
+    public ApplicantType getApplicantType(AsylumCase asylumCase) {
+        return asylumCase
+            .read(FTPA_APPLICANT_TYPE, ApplicantType.class)
```

Comment:
How this is work when we have second FTPA appeal?

Reply:
It will always return the applicant